### PR TITLE
CORE-827: Upgrade eslint, set environment for es2020

### PR DESCRIPTION
**Pull request by kuguarpwnz**

### CODE REVIEW

**Review by sturchaninov — APPROVED** [review]
🔥

File: lib/defaults.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   'env': {                           // http://eslint.org/docs/user-guide/configuring.html#specifying-environments
     'browser': true,                 // browser global variables
     'node': true,                    // Node.js global variables and Node.js-specific rules
-    'es6': true
+    'es2020': true
```
Comment (svirtryu) [inline lib/defaults.js:8]: I don't know if we should do this because I don't think we support transpiling ES2020 code

Comment (kuguarpwnz) [reply lib/defaults.js:8]: that's why it's a major version change

Comment (kuguarpwnz) [reply lib/defaults.js:8]: I need this to make eslint work with optional chaining `a?.b`

**Review by svirtryu — APPROVED** [review]

**Pull request merged**